pkg/app/mssql.go

```diff
@@ -0,0 +1,366 @@
+	m.deployment = deployment
+
+	// Create Service
+	serviceObj, err := m.getServiceObj()
+	if err != nil {
+		return err
+	}
+	service, err := m.cli.CoreV1().Services(namespace).Create(ctx, serviceObj, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("Service with name " + service.Name + " created successfully")
+	m.service = service
+
+	return err
+}
+
+func (m *MssqlDB) IsReady(ctx context.Context) (bool, error) {
+	log.Print("Waiting for the mssql deployment to be ready.", field.M{"app": m.name})
+	ctx, cancel := context.WithTimeout(ctx, mssqlWaitTimeout)
+	defer cancel()
+
+	err := kube.WaitOnDeploymentReady(ctx, m.cli, m.namespace, m.deployment.Name)
+	if err != nil {
+		return false, err
+	}
+	log.Print("Application instance is ready.", field.M{"app": m.name})
+	return true, nil
+}
+
+func (m *MssqlDB) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "deployment",
+		Name:      "mssql-deployment",
+		Namespace: m.namespace,
+	}
+}
+
+func (m *MssqlDB) Uninstall(ctx context.Context) error {
+	// Delete PVC
+	err := m.cli.CoreV1().PersistentVolumeClaims(m.namespace).Delete(ctx, m.pvc.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("PVC deleted successfully")
+
+	// Delete Deployment
+	err = m.cli.AppsV1().Deployments(m.namespace).Delete(ctx, m.deployment.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("Deployment deleted successfully")
+
+	// Delete Service
+	err = m.cli.CoreV1().Services(m.namespace).Delete(ctx, m.service.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("Service deleted successfully")
+
+	//Delete Secret
+	err = m.cli.CoreV1().Secrets(m.namespace).Delete(ctx, m.secret.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("Secret deleted successfully")
+	return nil
+}
+
+func (m *MssqlDB) Ping(ctx context.Context) error {
+	log.Print("Pinging database")
+	count := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"SELECT name FROM sys.databases WHERE name NOT IN ('master','model','msdb','tempdb')\" -b -s \",\" -h -1"
+
+	loginMssql := []string{"sh", "-c", count}
+	_, stderr, err := m.execCommand(ctx, loginMssql)
+	if err != nil {
+		return errors.Wrapf(err, "Error while Pinging the database %s", stderr)
+	}
+	log.Print("Ping to the application was success.", field.M{"app": m.name})
+	return err
+}
+
+func (m *MssqlDB) Insert(ctx context.Context) error {
+	log.Print("Adding entry to database")
+	insert := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"USE test; INSERT INTO Inventory VALUES (1, 'banana', 150)\""
+
+	insertQuery := []string{"sh", "-c", insert}
+	_, stderr, err := m.execCommand(ctx, insertQuery)
+	if err != nil {
+		return errors.Wrapf(err, "Error while inserting data into table %s", stderr)
```

Comment:
same problem here, table %s, looks like we are printing table name instead of error.

Reply:
applies at a lot of other places

Reply:
sure. I will add visible separator over there. We are following same pattern in other apps.